packages/data/src/world/mq/gerudo_training_grounds_mq.yml

```diff
@@ -69,38 +196,90 @@
-    "Gerudo Training Grounds Water Room": "silver_rupees_gtg_lava_mq && event(GTG_LAVA_TARGETS) && (can_hookshot || has_hover_boots || trick(OOT_MQ_GTG_FLAMES))"
+    "Gerudo Training Grounds Stalfos Room After Silver Block": "true"
   events:
-    GTG_LAVA_HAMMER: "can_hammer"
-    GTG_LAVA_TARGETS: "silver_rupees_gtg_lava_mq"
-  locations:
-    "MQ Gerudo Training Grounds Spinning Statue Chest": "can_use_bow"
-    "MQ Gerudo Training Grounds Torch Slug Room Clear Chest": "soul_iron_knuckle && soul_enemy(SOUL_ENEMY_TORCH_SLUG)"
-    "MQ Gerudo Training Grounds Torch Slug Room Switch Chest": "has_ranged_weapon" #You are no longer guaranteed a hook due to silver rupee shuffle.
-    "MQ Gerudo Training Grounds Maze Right Side Middle Chest": "event(GTG_LAVA_HAMMER) && (can_longshot || (can_hookshot && (has_fire_arrows || (has_hover_boots && event(GTG_LAVA_TARGETS)) || trick(OOT_MQ_GTG_FLAMES))))"
-    "MQ Gerudo Training Grounds Maze Right Side Right Chest": "event(GTG_LAVA_HAMMER) && (can_longshot || (can_hookshot && (has_fire_arrows || (has_hover_boots && event(GTG_LAVA_TARGETS)) || trick(OOT_MQ_GTG_FLAMES))))"
-    "MQ Gerudo Training Grounds Ice Arrows Chest": "event(GTG_ICE_ARROWS)"
-    "MQ Gerudo Training Grounds SR Lava Back-Right": "has_fire_arrows"
-    "MQ Gerudo Training Grounds SR Lava Center": "has_fire_arrows"
-    "MQ Gerudo Training Grounds Wonder Item Eye Statue": "has_hover_boots"
+    GTG_MQ_STALFOS_SIDE_CLEAR: "soul_enemy(SOUL_ENEMY_SPIKE) && soul_freezard && (can_use_sword_master || can_use_sword_razor || (has(SWORD_BIGGORON) && (is_adult || (is_child && age_sword_adult))))" #Kokiri Sword is too weak against freezards
+  locations:
+    "MQ Gerudo Training Grounds Silver Block Room Chest": "event(GTG_MQ_STALFOS_SIDE_CLEAR)"
+"Gerudo Training Grounds Stalfos Room Ledge Before Red Ice":
+  dungeon: GTG
+  exits:
+    "Gerudo Training Grounds Stalfos Room Ledge After Red Ice": "has_blue_fire"
+    "Gerudo Training Grounds Stalfos Room": "has_lens"
+"Gerudo Training Grounds Stalfos Room Ledge After Red Ice":
+  dungeon: GTG
+  exits:
+    "Gerudo Training Grounds Stalfos Room Ledge Before Red Ice": "has_blue_fire" #Might not be solid from other side.
+    "Gerudo Training Grounds Spinning Statue Room Upper": "event(GTG_MQ_STALFOS_CLEAR)"
+"Gerudo Training Grounds Spinning Statue Room Upper":
+  dungeon: GTG
+  exits:
+    "Gerudo Training Grounds Stalfos Room Ledge After Red Ice": "true"
+    "Gerudo Training Grounds Spinning Statue Room Upper Side Door": "event(GTG_MQ_SPINNING_STATUE_CRYSTAL)"
+    "Gerudo Training Grounds Spinning Statue Room Lower": "true"
+    "Gerudo Training Grounds Spinning Statue Room Statue Middle": "has_hover_boots || can_longshot || hookshot_anywhere"
+"Gerudo Training Grounds Spinning Statue Room Upper Side Door":
```

Comment:
Should be able to hit the crystal from up on this platform and therefore not need to jump down (granted, a void will put you back up anyway)